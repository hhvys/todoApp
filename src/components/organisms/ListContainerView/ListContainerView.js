import React from 'react';
import RowComponent from "../../molecules/RowComponent/RowComponent";
import {INBOX_ID, STARRED_ID} from "../../../actions/actionTypes";
import {INBOX, STAR} from "../../atoms/Icons/constants";
import styles from './ListContainerView.mod.scss';

class ListContainerView extends React.Component {

	renderTabs = ({
												 activeTab,
												 tabs,
												 onClick,
												 onHeaderSymbolClick,
												 onFooterSymbolClick,
												 headerSymbol,
												 footerSymbol
											 }) => {
		return (
			tabs.map(tab => {
				if (!tab.footerContent && tab.tabId === STARRED_ID)
					return null;
				return (
					<RowComponent
						className={styles.tab}
						key={tab.tabId}
						onClick={() => onClick(tab.tabId)}
						onHeaderSymbolClick={onHeaderSymbolClick}
						onFooterSymbolClick={() => onFooterSymbolClick(tab.tabId)}
						headerSymbol={
							tab.tabId === INBOX_ID ?
								INBOX :
								tab.tabId === STARRED_ID ?
									STAR :
									headerSymbol
						}

						footerSymbol={
							activeTab === tab.tabId &&
							activeTab !== INBOX_ID &&
							activeTab !== STARRED_ID &&
							footerSymbol
						}

						footerContent={
							tab.footerContent > 0 && tab.footerContent
						}
						mainContent={tab.tabName}
						active={activeTab === tab.tabId}
						headerClass={
							`${styles.logo} ${
								tab.tabId === INBOX_ID ?
									styles.inboxLogo :
									tab.tabId === STARRED_ID ?
										styles.starLogo :
										styles.listLogo
								}`
						}
						footerClass={styles.footerLogo}
					/>
				);
			})
		);
	};

	render() {
		const {className} = this.props;
		return (
			<div className={`${className ? className : ''} flex-column justify-content-start align-items-center`}>
				{
					this.renderTabs(this.props)
				}
			</div>
		);
	}
}

export default ListContainerView;
