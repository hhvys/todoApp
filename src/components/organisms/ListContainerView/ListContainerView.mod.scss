@import "../../../constants.mod";
.tab {
  height: 38px;
}

.icon {
  width: 30px;
}

.inboxLogo {
  fill: rgb(50, 138, 214);
}

.starLogo {
  fill: red;
}

.listLogo {
  fill: #b9b9b9;
}

.footerLogo {
  fill: $icon-primary-color;
  opacity: 0.6;
  margin-right: 5px;
}

.footerLogo:hover {
  opacity: 1;
}

.active {
  background-color: #E1EEFA;
}
