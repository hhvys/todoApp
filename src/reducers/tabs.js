import {
	ADD_TAB,
	CHANGE_TAB_NAME,
	ADD_TODO,
	TOGGLE_TODO,
	DELETE_TAB,
	COPY_TAB,
	STAR_TOGGLE_TODO, ADD_STARRED_TODO, ACTIVE_TODO
} from '../actions/actionTypes';

const todos = (state = [], action) => {
	switch (action.type) {
		case ADD_TODO:
			return (
				[
					...state,
					{
						todoId: action.todoId,
						text: action.text,
						completed: false,
						star: false,
						completedTime: undefined,
						createdTime: new Date(),
						active: false
					}
				]);
		case ADD_STARRED_TODO:
			return (
				[
					...state,
					{
						todoId: action.todoId,
						text: action.text,
						completed: false,
						star: true,
						completedTime: undefined,
						createdTime: new Date(),
						active: false
					}
				]);
		case TOGGLE_TODO:
			return state.map(todo => (
				todo.todoId === action.todoId ?
					{
						...todo,
						completed: !todo.completed,
						completedTime: new Date()
					} :
					todo
			));
		case STAR_TOGGLE_TODO:
			return state.map(todo => (
				todo.todoId === action.todoId ?
					{
						...todo,
						star: !todo.star
					} :
					todo
			));
		case ACTIVE_TODO:
			return state.map(todo => (
				todo.todoId === action.todoId ?
					{
						...todo,
						active: true
					} :
					{
						...todo,
						active: false
					}
			));
		default:
			return state;
	}
};

const initialState = [
	{
		tabId: 0,
		tabName: 'Inbox',
		todos: []
	},
	{
		tabId: 1,
		tabName: 'Starred',
		todos: []
	}
];

function tabs(state = initialState, action) {
	switch (action.type) {
		case ADD_TAB:
			return [
				...state,
				{
					tabId: action.tabId,
					tabName: action.tabName,
					todos: []
				}
			];
		case CHANGE_TAB_NAME:
			return state.map(tab => (
				tab.tabId === action.tabId ?
					{
						tabId: tab.tabId,
						tabName:
						action.tabName,
						todos: tab.todos
					} :
					tab
			));
		case COPY_TAB:
			let fromTab =
				state.find(tab => tab.tabId === action.fromId);
			return (
				[
					...state,
					{
						tabId: action.toId,
						tabName: fromTab.tabName + " Copy",
						todos: [...fromTab.todos]
					}
				]
			);
		case DELETE_TAB:
			return state.filter(tab =>
				tab.tabId !== action.tabId);
		case ADD_TODO:
		case ADD_STARRED_TODO:
		case TOGGLE_TODO:
		case STAR_TOGGLE_TODO:
		case ACTIVE_TODO:
			return state.map(tab => (
				tab.tabId === action.tabId ?
					{
						...tab,
						todos: todos(tab.todos, action)
					} : tab
			));
		default:
			return state;
	}
}

export default tabs;