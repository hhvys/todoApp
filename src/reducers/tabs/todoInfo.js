import {
	ACTIVE_TODO,
	ADD_STARRED_TODO,
	ADD_TODO,
	COPY_TAB,
	STAR_TOGGLE_TODO,
	TOGGLE_TODO
} from "../../actions/actionTypes";

const todo = (state = {}, action) => {
	switch (action.type) {
		case ADD_TODO:
			return {
				tabId: action.tabId,
				todoId: action.todoId,
				text: action.text,
				completed: false,
				star: false,
				completedTime: undefined,
				createdTime: new Date(),
				active: false
			};
		case ADD_STARRED_TODO:
			return {
				tabId: action.tabId,
				todoId: action.todoId,
				text: action.text,
				completed: false,
				star: true,
				completedTime: undefined,
				createdTime: new Date(),
				active: false
			};
		case TOGGLE_TODO:
			return {
				...state,
				completed: !state.completed,
				completedTime: new Date()
			};
		case STAR_TOGGLE_TODO:
			return {
				...state,
				star: !state.star
			};
		default:
			return state;
	}
};

function todoInfo(state = {}, action) {
	switch (action.type) {
		case STAR_TOGGLE_TODO:
		case ADD_TODO:
		case ADD_STARRED_TODO:
		case TOGGLE_TODO:
		case ACTIVE_TODO:
			return {
				...state,
				[action.todoId]: todo(state[action.todoId], action)
			};
		case COPY_TAB:
			return {...state, ...(action.todos)};
		default:
			return state;
	}
}

export default todoInfo;

export function getTodoInfo(state, todos) {
	if (Array.isArray(todos)) {
		return todos.map(todo => (
			state.tabs.todoInfo[todo]
		))
	}
	return state.tabs.todoInfo[todos];
}
