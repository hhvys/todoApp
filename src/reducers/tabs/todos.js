import {
	ADD_TODO,
	ADD_STARRED_TODO,
	STAR_TOGGLE_TODO,
	SORT_BY, CHANGE_SORT
} from "../../actions/actionTypes";

const todos = (state = [], action, todoInfo) => {
	switch (action.type) {
		case CHANGE_SORT:
			return state.sort((a, b) => {
				a = todoInfo[a];
				b = todoInfo[b];
				switch (action.sortBy) {
					case SORT_BY.SORT_PRIORITY:
						if (a.star && !b.star)
							return -1;
						if (!a.star && b.star)
							return 1;
						return 0;

					case SORT_BY.SORT_CREATION:
						if (a.createdTime > b.createdTime)
							return -1;
						return 1;

					case SORT_BY.SORT_ALPHA:
						if (a.text.toLowerCase() < b.text.toLowerCase())
							return -1;
						return 1;
				}
			});
		case ADD_TODO:
			return (
				[
					...state,
					action.todoId
				]);
		case ADD_STARRED_TODO:
			return (
				[
					...state,
					action.todoId
				]);
		case STAR_TOGGLE_TODO:
			const index = state.findIndex(todoId => action.todoId === todoId);
			if (index === -1) {
				return [
					...state,
					action.todoId
				];
			}
			else {
				const updatedState = state;
				updatedState.splice(index, 1);
				return updatedState;
			}
		default:
			return state;
	}
};

export default todos;
