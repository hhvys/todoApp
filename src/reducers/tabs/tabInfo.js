import {
	ADD_TODO,
	ADD_STARRED_TODO,
	STAR_TOGGLE_TODO,
	SORT_BY,
	CHANGE_SORT,
	ADD_TAB,
	INBOX_ID,
	STARRED_ID,
	COPY_TAB,
	CHANGE_TAB_NAME,
	TOGGLE_VISIBILITY_FILTER,
	DELETE_TAB,
	TOGGLE_TODO
} from "../../actions/actionTypes";
import {getTabs} from "./tabs";

/**
 *
 * @param todos - array of to-do ids.
 * @param sortBy
 * @param todoById
 * @returns {*} - sorted array of to-do ids.
 */

const stableSortTodos = (todos, sortBy, todoById) => {
	//For stable sort
	const indexOf = {};
	todos.forEach((todo, index) => {
		indexOf[todo] = index;
	});

	const compareByIndex = (a, b) => {
		if (indexOf[a.todoId] < indexOf[b.todoId])
			return -1;
		return 1;
	};
	return todos.sort((todo1, todo2) => {
		const a = todoById[todo1];
		const b = todoById[todo2];
		switch (sortBy) {
			case SORT_BY.SORT_PRIORITY:
				if (a.star && !b.star)
					return -1;
				if (!a.star && b.star)
					return 1;
				return compareByIndex(a, b);

			case SORT_BY.SORT_CREATION:

				if (a.createdTime > b.createdTime)
					return -1;
				if (a.createdTime < b.createdTime)
					return 1;
				return compareByIndex(a, b);

			case SORT_BY.SORT_ALPHA:
				if (a.text.toLowerCase() < b.text.toLowerCase())
					return -1;
				if (a.text.toLowerCase() > b.text.toLowerCase())
					return 1;
				return compareByIndex(a, b);
		}
	});
};

const todos = (state = [], action, todoById) => {
	switch (action.type) {

		case CHANGE_SORT:
			const newState = [...state];
			return stableSortTodos(newState, action.sortBy, todoById);

		case ADD_TODO:
		case ADD_STARRED_TODO:
			return (
				[
					...state,
					action.todoId
				]);

		case STAR_TOGGLE_TODO:

			/*
				Add to-do id to state array if to-do id is not present and remove if present.
			 */

			const index = state.findIndex(todoId => action.todoId === todoId);
			if (index === -1) {
				return [
					...state,
					action.todoId
				];
			}
			const updatedState = state;
			updatedState.splice(index, 1);
			return updatedState;

		default:
			return state;
	}
};

export const initialState = [
	{
		tabId: INBOX_ID,
		tabName: 'Inbox',
		todos: [],
		starredTodos: [],
		inCompletedTodos: 0
	},
	{
		tabId: STARRED_ID,
		tabName: 'Starred',
		todos: [],
		starredTodos: [],
		inCompletedTodos: 0
	}
];

function tabInfo(state = initialState, action, todoById) {
	let newState;
	switch (action.type) {
		case CHANGE_SORT:
			return state.map(tab => {
				if (tab.tabId !== action.tabId)
					return tab;
				return ({
					...tab,
					todos: todos(tab.todos, action, todoById),
					starredTodos: todos(tab.starredTodos, action, todoById)
				})
			});

		case ADD_TAB:
			return [
				...state,
				{
					tabId: action.tabId,
					tabName: action.tabName,
					todos: [],
					starredTodos: [],
					inCompletedTodos: 0
				}
			];

		case CHANGE_TAB_NAME:
			return state.map(tab => (
				tab.tabId === action.tabId ?
					{
						...tab,
						tabId: tab.tabId,
						tabName: action.tabName,
					} :
					tab
			));

		case COPY_TAB:

			newState = state.map(tab => {
				if (tab.tabId === STARRED_ID) {
					let change = 0;
					Object
						.keys(action.todos)
						.forEach(todo => {
							if (!action.todos[todo].completed && action.todos[todo].star)
								change += 1;
						});
					return {
						...tab,
						inCompletedTodos: tab.inCompletedTodos + change
					}
				}
				return tab;
			});

			return (
				[
					...newState,
					{
						tabId: action.toId,
						tabName: action.tabName,
						todos: Object
							.keys(action.todos)
							.map(key => action.todos[key].todoId),
						starredTodos: [...action.starredTodos],
						inCompletedTodos: action.inCompletedTodos
					}
				]
			);
		case DELETE_TAB:
			newState = state.map(tab => {
				if (tab.tabId === STARRED_ID) {
					let change = 0;
					action
						.todos
						.forEach(todo => {
							if (!todoById[todo].completed && todoById[todo].star)
								change += 1;
						});
					return {
						...tab,
						inCompletedTodos: tab.inCompletedTodos - change
					}
				}
				return tab;
			});
			return newState.filter(tab =>
				tab.tabId !== action.tabId);
		case TOGGLE_VISIBILITY_FILTER:
			return state.map(tab => (
				tab.tabId === action.tabId ?
					{
						...tab,
						showCompletedTodo: !tab.showCompletedTodo
					} :
					tab
			));
		case STAR_TOGGLE_TODO:
			return state.map(tab => {
				if (tab.tabId === STARRED_ID) {
					let change = action.star ? -1 : +1;
					if (action.completed)
						change = 0;
					return {
						...tab,
						inCompletedTodos: tab.inCompletedTodos + change
					};
				}
				if (tab.tabId === action.tabId) {
					return {
						...tab,
						starredTodos: todos(tab.starredTodos, action),
					};
				}
				return tab;
			});
		case ADD_TODO:
			return state.map(tab => (
				tab.tabId === action.tabId ?
					{
						...tab,
						todos: todos(tab.todos, action),
						inCompletedTodos: tab.inCompletedTodos + 1
					} :
					tab
			));
		case ADD_STARRED_TODO:
			return state.map(tab => {
				if (tab.tabId === action.tabId)
					return {
						...tab,
						starredTodos: todos(tab.starredTodos, action),
						todos: todos(tab.todos, action),
						inCompletedTodos: tab.inCompletedTodos + 1
					};
				if (tab.tabId === STARRED_ID)
					return {
						...tab,
						inCompletedTodos: tab.inCompletedTodos + 1
					};
				return tab
			});
		case TOGGLE_TODO:
			const changeStarredTab = action.star;

			return state.map(tab => {
				if (tab.tabId === action.tabId || (changeStarredTab && tab.tabId === STARRED_ID)) {
					const change = !action.completed ? -1 : +1;
					return {
						...tab,
						inCompletedTodos: tab.inCompletedTodos + change
					};
				}
				return tab;
			});
		default:
			return state;
	}
}

export const getTabInfo = (state, tabId) => {
	state = getTabs(state);
	if (typeof tabId === 'undefined')
		return state.tabInfo;
	return state
		.tabInfo
		.find(tab => tab.tabId === tabId);
};

export default tabInfo;
