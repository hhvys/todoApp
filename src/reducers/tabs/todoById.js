import {
	ADD_STARRED_TODO,
	ADD_TODO,
	COPY_TAB, DELETE_TAB,
	STAR_TOGGLE_TODO,
	TOGGLE_TODO
} from "../../actions/actionTypes";
import {getTabs} from "./tabs";

/**
 *
 * @param state - Object containing information for particular todo
 * @param action
 * @returns {*} - Updated object containing information for todo
 */

const updateTodoInfo = (state = {}, action) => {
	switch (action.type) {
		case ADD_TODO:
			return {
				tabId: action.tabId,
				todoId: action.todoId,
				text: action.text,
				createdTime: new Date(),
			};
		case ADD_STARRED_TODO:
			return {
				tabId: action.tabId,
				todoId: action.todoId,
				text: action.text,
				star: true,
				createdTime: new Date(),
			};
		case TOGGLE_TODO:
			return {
				...state,
				completed: !state.completed,
				completedTime: new Date()
			};
		case STAR_TOGGLE_TODO:
			return {
				...state,
				star: !state.star
			};
		default:
			return state;
	}
};

function todoById(state = {}, action) {
	switch (action.type) {
		case STAR_TOGGLE_TODO:
		case ADD_TODO:
		case ADD_STARRED_TODO:
		case TOGGLE_TODO:
			state[action.todoId] = updateTodoInfo(state[action.todoId], action);
			return state;
		case COPY_TAB:
			return {...state, ...(action.todos)};
		case DELETE_TAB:
			const newState = {...state};
			action.todos.forEach(todo => delete newState[todo]);
			return newState;
		default:
			return state;
	}
}

export default todoById;



/**
 * Selector from todoById
 * @param state - full state
 * @param todos - single todoId or array of todoIds
 * @returns {*} - object or array of object with full todoInfo
 */

export function getTodoById(state, todos) {
	state = getTabs(state);
	if (Array.isArray(todos)) {
		return todos.map(todo => (
			state.todoById[todo]
		))
	}
	return state.todoById[todos];
}
