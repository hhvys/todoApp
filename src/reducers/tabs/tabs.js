import {combineReducers} from 'redux';
import todo from './todo';
import todos from './todos';
import {
	ADD_TAB,
	CHANGE_TAB_NAME,
	ADD_TODO,
	TOGGLE_TODO,
	DELETE_TAB,
	COPY_TAB,
	STAR_TOGGLE_TODO,
	ADD_STARRED_TODO,
	ACTIVE_TODO,
	TOGGLE_VISIBILITY_FILTER,
	INBOX_ID,
	STARRED_ID
} from '../../actions/actionTypes';


const initialState = [
	{
		tabId: INBOX_ID,
		tabName: 'Inbox',
		todos: [],
		starredTodos: []
	},
	{
		tabId: STARRED_ID,
		tabName: 'Starred',
		todos: [],
		starredTodos: []
	}
];

function tabInfo(state = initialState, action) {
	switch (action.type) {
		case ADD_TAB:
			return [
				...state,
				{
					tabId: action.tabId,
					tabName: action.tabName,
					todos: [],
					showCompletedTodo: true,
					starredTodos: []
				}
			];
		case CHANGE_TAB_NAME:
			return state.map(tab => (
				tab.tabId === action.tabId ?
					{
						...tab,
						tabId: tab.tabId,
						tabName: action.tabName,
					} :
					tab
			));
		case COPY_TAB:
			let fromTab =
				state.find(tab => tab.tabId === action.fromId);
			return (
				[
					...state,
					{
						tabId: action.toId,
						tabName: fromTab.tabName + " Copy",
						todos: [...fromTab.todos],
						starredTodos: [...fromTab.starredTodos]
					}
				]
			);
		case DELETE_TAB:
			return state.filter(tab =>
				tab.tabId !== action.tabId);
		case TOGGLE_VISIBILITY_FILTER:
			return state.map(tab => (
				tab.tabId === action.tabId ?
					{
						...tab,
						showCompletedTodo: !tab.showCompletedTodo
					} :
					tab
			));
		case STAR_TOGGLE_TODO:
			return state.map(tab => (
				tab.tabId === action.tabId ?
					{
						...tab,
						starredTodos: todos(tab.starredTodos, action)
					} : tab
			));
		case ADD_TODO:
			return state.map(tab => (
				tab.tabId === action.tabId ?
					{
						...tab,
						todos: todos(tab.todos, action)
					} : tab
			));
		case ADD_STARRED_TODO:
			return state.map(tab => (
				tab.tabId === action.tabId ?
					{
						...tab,
						starredTodos: todos(tab.starredTodos, action),
						todos: todos(tab.todos, action)
					} : tab
			));
		default:
			return state;
	}
}

function todoInfo(state = {}, action) {
	switch (action.type) {
		case STAR_TOGGLE_TODO:
		case ADD_TODO:
		case ADD_STARRED_TODO:
		case TOGGLE_TODO:
		case ACTIVE_TODO:
			return {
				...state,
				[action.todoId]: todo(state[action.todoId], action)
			};
		default:
			return state;
	}
}

const tabs = combineReducers({
	todoInfo,
	tabInfo
});

export default tabs;

export function getTabs(state) {
	const tabs = state.tabs.tabInfo.map(tab => {
		const todos = tab.todos.map(todo => (
			state.tabs.todoInfo[todo]
		));
		const starredTodos = tab.starredTodos.map(todo => (
			state.tabs.todoInfo[todo]
		));
		return {
			...tab,
			todos,
			starredTodos
		}
	});
	return tabs;
}
