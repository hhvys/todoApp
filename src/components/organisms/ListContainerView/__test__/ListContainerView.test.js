import React from 'react';
import ListContainerView from '../ListContainerView';
import {initialState} from "../../../../reducers/tabs/tabInfo";
import starredState from './starredTabInput';
import starredState2 from './starredInput2';

describe('ListContainerView', () => {

	let props;
	let mountedListContainer;

	const listContainerView = () => {
		if (!mountedListContainer) {
			mountedListContainer = shallow(
				<ListContainerView {...props}/>
			);
		}
		return mountedListContainer;
	};

	beforeEach(() => {
		props = {
			className: "className",
			activeTab: undefined,
			tabs: initialState,
			onClick: undefined,
			onHeaderSymbolClick: undefined,
			onFooterSymbolClick: undefined,
			headerSymbol: undefined,
			footerSymbol: undefined,
		};
		mountedListContainer = undefined;
	});

	it('should render on initalState', () => {
		const wrapperDiv = listContainerView().find('div').first();
		expect(toJSON(wrapperDiv)).toMatchSnapshot();
	});

	it('should have number of RowComponent = tabs.length - 1 if no uncompleted starredTodos are available', () => {
		props.tabs = starredState2;
		const RowComponents = listContainerView().find('RowComponent');
		expect(RowComponents.length).toBe(props.tabs.length - 1);
	});

	it('should have number of RowComponent = tabs.length if uncompleted starredTodos are available', () => {
		props.tabs = starredState;
		const RowComponents = listContainerView().find('RowComponent');
		expect(RowComponents.length).toBe(props.tabs.length);
	});

});
