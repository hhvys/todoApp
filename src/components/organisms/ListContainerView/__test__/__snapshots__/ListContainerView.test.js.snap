// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListContainerView should render on initalState 1`] = `
<div
  className="className flex-column justify-content-start align-items-center"
>
  <RowComponent
    className="tab "
    footerClass="footerLogo"
    footerClickArgs={
      Array [
        0,
      ]
    }
    footerIcon={false}
    headerClass="icon inboxLogo"
    headerIcon="INBOX"
    key="0"
    mainContent="Inbox"
    onClick={[Function]}
  />
</div>
`;
